lib/Sub/Future.pm

```diff
@@ -53,21 +53,24 @@ until the value is ready.
 
 sub future(&) {
     my ($code) = @_;
-    
+    my ($pid, $reader, $writer);
+
+    # open bidirectional pipe
+    pipe($reader, $writer) || die "Can't open pipe: $!\n";
+
     # create a new process to handle the spawned calculation
-    my $pid = open my $fh, '-|';
+    $pid = fork();
     die "Unable to fork: $!\n" if not defined $pid;
-
+    
     # the child process performs the calculation
     if ( not $pid ) {
         my $value = $code->();
-        my $frozen = freeze { returned => $value };
-        print $frozen;
+        print $writer freeze { returned => $value };
```

Comment:
Please keep freeze and print on separate lines. It facilitates debugging later. It also makes this diff smaller to highlight the functional changes.